autogen/oai/client.py

```diff
@@ -269,7 +269,7 @@ def _completions_create(self, client, params):
         completions = client.chat.completions if "messages" in params else client.completions
         # If streaming is enabled, has messages, and does not have functions, then
         # iterate over the chunks of the response
-        if params.get("stream", False) and "messages" in params and "functions" not in params:
+        if params.get("stream", False) and "messages" in params:
```

Comment:
This change is dangerous.
If we want to support streaming with functions, we need to handle function chunks.

Reply:
The response when using streaming with functions is something like this:
```Choice(
	delta=ChoiceDelta(
		content=None,
		function_call=ChoiceDeltaFunctionCall(
			arguments='',
			name='get_current_weather'),
			role='assistant', tool_calls=None),
			finish_reason=None, index=0
)
```
As you can see, content is None. We have to handle the 'function_call' field instead.
If we remove the "...and functions not in param", we will be looking for content when we should be looking for 'function_call'.

Reply:
hey @Alvaromah fair point but I believe line 284 checks the `ChoiceDelta` to ensure `choice.delta.content` is None and that `function_call` is present, which, if I understand correctly, would then handle that use case